src/mono/sample/wasm/blazor-frame/wwwroot/frame.js

```diff
@@ -0,0 +1,43 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+"use strict";
+
+class FrameApp {
+    async init({ getAssemblyExports }) {
+        const exports = await getAssemblyExports("blazor.dll");
+        exports.BrowserBench.FrameApp.ReachedManaged();
+    }
+
+    reachedCallback() {
+        if (window.parent != window) {
+            window.parent.resolveAppStartEvent("reached");
+        }
+    }
+}
+
+let mute = false;
+try {
+    globalThis.frameApp = new FrameApp();
+    globalThis.frameApp.ReachedCallback = globalThis.frameApp.reachedCallback.bind(globalThis.frameApp);
+    if (window.parent != window) {
+        window.addEventListener("pageshow", event => { window.parent.resolveAppStartEvent("pageshow"); })
+    }
+
+    window.muteErrors = () => {
+        mute = true;
+    }
+
+    await Blazor.start();
```

Comment:
This call already starts .NET, it calls main inside of it 
- https://github.com/dotnet/aspnetcore/blob/4a9118c674a798aaf6379b4b7b2d8787bc688f96/src/Components/Web.JS/src/Boot.WebAssembly.ts#L29
- https://github.com/dotnet/aspnetcore/blob/4a9118c674a798aaf6379b4b7b2d8787bc688f96/src/Components/Web.JS/src/Boot.WebAssembly.Common.ts#L143

Maybe we should pass `configureRuntime` to `Blazor.start` and register to `onDotnetReady` in `withModuleConfig`?

```
Blazor.start({
    configureRuntime(api => api.withModuleConfig({
        onDotnetReady: window.parent.resolveAppStartEvent("onDotnetReady");
    })
});
```

Reply:
We don't use onDotnetReady anymore, so I removed it in blazor-frame and also in the browser-bench itself.

Reply:
Okay, I'm just saying this is more like "reach managed" than "dotnet ready"

Reply:
after offsite discussion: I missed the mention of Main being called inside `Blazor.Start`. It makes sense then to use `onDotnetReady`. It should be close enough to how we measure the existing startup.

Reply:
In the end I made it call `BrowserBench.FrameApp.ReachedManaged` from the `Main`. I wanted to preserve JS/managed/JS roundtrip and at the time `onDotnetReady` is called, the `Blazor.runtime` is not yet assigned.

 This way we can even go further to the past with the measurements.